### ISSUE TITLE
qgis_sipify: fatal: not a git repository (or any of the parent directories): .git

### ISSUE THREAD

**Issue by strk** [qgis/QGIS#32783]
A snippet from `make check` merged stdout/stderr, running from an out-of-source build tree:
```
578/583 Test #576: qgis_spelling .................................   Passed    0.00 sec
        Start 577: qgis_shellcheck
579/583 Test #577: qgis_shellcheck ...............................***Failed    0.00 sec
fatal: not a git repository (or any of the parent directories): .git
/usr/src/qgis/qgis-do-not-override-source-files/tests/code_layout/test_shellcheck.sh: line 7: pushd: no other directory

        Start 578: qgis_sipify
580/583 Test #578: qgis_sipify ...................................***Failed    0.00 sec
fatal: not a git repository (or any of the parent directories): .git
/usr/src/qgis/qgis-do-not-override-source-files/tests/code_layout/test_sipify.sh: line 7: pushd: no other directory

        Start 579: qgis_sip_include
581/583 Test #579: qgis_sip_include ..............................***Failed    0.00 sec
fatal: not a git repository (or any of the parent directories): .git
fatal: not a git repository (or any of the parent directories): .git
/usr/src/qgis/qgis-do-not-override-source-files/tests/code_layout/test_sip_include.sh: line 6: pushd: no other directory

        Start 580: qgis_sip_uptodate
582/583 Test #580: qgis_sip_uptodate .............................***Failed    0.00 sec
fatal: not a git repository (or any of the parent directories): .git
/usr/src/qgis/qgis-do-not-override-source-files/tests/code_layout/test_sipfiles_uptodate.sh: line 13: pushd: no other directory

        Start 581: qgis_doxygen_order
583/583 Test #581: qgis_doxygen_order ............................   Passed    0.05 sec
```

It looks like there's some wrong assumption in some script ?

**Comment by strk** [comment]
Can be reproduced with: 
> $ ctest -R qgis_shellcheck --output-on-failure

I guess the assumption is that the build tree is under the source tree ?

**Comment by strk** [comment]
Bug is fixed by merging #32784
